test/Test/Iris/Tool.hs

```diff
@@ -2,40 +2,56 @@ module Test.Iris.Tool (toolSpec) where
 
 import Data.Text (Text)
 import Data.Version (Version, makeVersion, parseVersion)
-import Test.Hspec (Spec, describe, it, shouldReturn, shouldSatisfy)
+import Test.Hspec (Spec, describe, it, shouldReturn, shouldSatisfy, shouldThrow)
 import Text.ParserCombinators.ReadP (readP_to_S)
 
-import Iris.Tool (Tool (..), ToolCheckResult (..), ToolCheckError (..), ToolSelector (..), checkTool,
-                  defaultToolSelector)
+import Iris.Tool (Tool (..), ToolCheckResult (..), ToolCheckError (..), ToolSelector (..), checkTool, ToolCheckException(..),
+                  defaultToolSelector, need)
 
 import qualified Data.Text as Text
 
+ghc100 :: Tool
+ghc100 = "ghc"
+    { toolSelector = Just defaultToolSelector
+        { toolSelectorFunction = \version -> case getVersion version of
+                Nothing -> False
+                Just v  -> v >= makeVersion [100]
+        , toolSelectorVersionArg = Just "--numeric-version"
+        }
+    }
 
 toolSpec :: Spec
 toolSpec = describe "Tool" $ do
     it "should find 'curl'" $ do
-        checkTool () "curl" `shouldReturn` ToolOk
+        checkTool "curl" `shouldReturn` ToolOk
 
     it "shouldn't find 'xxx_unknown_executable'" $ do
-        checkTool () "xxx_unknown_executable"
+        checkTool "xxx_unknown_executable"
             `shouldReturn` ToolCheckError (ToolNotFound "xxx_unknown_executable")
 
     it "shouldn't find 'ghc' version 100" $ do
-        let tool :: Tool ()
-            tool = "ghc"
-                { toolSelector = Just defaultToolSelector
-                    { toolSelectorFunction = \_cmd version -> case getVersion version of
-                          Nothing -> False
-                          Just v  -> v >= makeVersion [100]
-                    , toolSelectorVersionArg = Just "--numeric-version"
-                    }
-                }
-
         let isToolWrongVersion :: ToolCheckResult -> Bool
             isToolWrongVersion (ToolCheckError (ToolWrongVersion _)) = True
             isToolWrongVersion _other               = False
 
-        checkTool () tool >>= (`shouldSatisfy` isToolWrongVersion)
+        checkTool ghc100 >>= (`shouldSatisfy` isToolWrongVersion)
+
+    it "should not fail when 'need'ing 'curl'" $ (need ["curl"] :: IO ())
```

Comment:
HLint has a warning here and thus CI is failing. I'll probably disable some of the annoying HLint warnings in the future but for now, let's just remove this extra `$` to make HLint happy
```suggestion
    it "should not fail when 'need'ing 'curl'" (need ["curl"] :: IO ())
```